alpha/apps/kaltura/lib/kSimuliveUtils.php

```diff
@@ -134,4 +134,25 @@ public static function getIsLiveCacheTime (LiveEntry $entry)
 		// conditional cache should expire when event start
 		return max($playableStartTime - $nowEpoch, self::SIMULIVE_SCHEDULE_MARGIN);
 	}
+
+	public static function createEntryServerNode(LiveEntry $entry, $endTime)
+	{
+		$expiry = $endTime - time();
+		$esn = EntryServerNodePeer::retrieveByEntryIdAndServerType($entry->getId(), EntryServerNodeType::LIVE_PRIMARY); // simulive always primary
+		if (!$esn && $expiry > 0)
```

Comment:
Maybe we can do:
if ($expiry > 0 && !EntryServerNodePeer::retrieveByEntryIdAndServerType($entry->getId(), EntryServerNodeType::LIVE_PRIMARY))
it little bit uglier but better performance